curp/src/client.rs

```diff
@@ -48,30 +50,23 @@ struct State {
     term: u64,
     /// When a new leader is set, notify
     leader_notify: Arc<Event>,
-    /// Send leader changes
-    leader_tx: broadcast::Sender<ServerId>,
 }
 
 impl State {
     /// Create the initial client state
     fn new() -> Self {
-        let (leader_tx, _) = broadcast::channel(1);
         Self {
             leader: None,
             term: 0,
             leader_notify: Arc::new(Event::new()),
-            leader_tx,
         }
     }
 
     /// Set the leader and notify a waiter
     fn set_leader(&mut self, id: ServerId) {
         debug!("client update its leader to {id}");
-        if self.leader.as_ref().map_or(true, |prev_id| prev_id != &id) {
-            let _ignored = self.leader_tx.send(id.clone()).ok(); // it's ok to have no receiver
-        }
         self.leader = Some(id);
-        self.leader_notify.notify(1);
+        self.leader_notify.notify(usize::MAX);
```

Comment:
If you mean notifying everyone, please correct the above comment.

Reply:
Done